applicationset/generators/pull_request.go

```diff
@@ -214,6 +214,9 @@ func (g *PullRequestGenerator) selectServiceProvider(ctx context.Context, genera
 	}
 	if generatorConfig.AzureDevOps != nil {
 		providerConfig := generatorConfig.AzureDevOps
+		if providerConfig.UseWorkloadIdentity {
+			return pullrequest.NewAzureDevOpsServiceWithWorkloadIdentity(providerConfig.API, providerConfig.Organization, providerConfig.Project, providerConfig.Repo, providerConfig.Labels)
```

Comment:
Have you considered just passing the `providerConfig` object instead of 5 different arguments?
All of them are using `providerConfig` values. 

WDYT?

Reply:
Thanks, updated